### ISSUE TITLE
entityfishing cannot be found

### ISSUE THREAD

**Issue by angelosalatino** [Lucaterre/spacyfishing#10]
Hi I am running the very first example in my environment and I bumped into this issue, which I can't figure out how to fix it.

Python returns this:

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/var/folders/rh/44gmh6sx6yv9pg9q4mc37dyd20hjyk/T/ipykernel_93919/2024306140.py in <module>
      1 nlp_model_en = spacy.load("en_core_web_sm")
      2 
----> 3 nlp_model_en.add_pipe("entityfishing")
      4 
      5 doc_en = nlp_model_en(text_en)

/opt/anaconda3/envs/spacyfishing/lib/python3.7/site-packages/spacy/language.py in add_pipe(self, factory_name, name, before, after, first, last, source, config, raw_config, validate)
    798                 config=config,
    799                 raw_config=raw_config,
--> 800                 validate=validate,
    801             )
    802         pipe_index = self._get_pipe_index(before, after, first, last)

/opt/anaconda3/envs/spacyfishing/lib/python3.7/site-packages/spacy/language.py in create_pipe(self, factory_name, name, config, raw_config, validate)
    653                 lang_code=self.lang,
    654             )
--> 655             raise ValueError(err)
    656         pipe_meta = self.get_factory_meta(factory_name)
    657         # This is unideal, but the alternative would mean you always need to

ValueError: [E002] Can't find factory for 'entityfishing' for language English (en). This usually happens when spaCy calls `nlp.create_pipe` with a custom component name that's not registered on the current language class. If you're using a Transformer, make sure to install 'spacy-transformers'. If you're using a custom component, make sure you've added the decorator `@Language.component` (for function components) or `@Language.factory` (for class components).

Available factories: attribute_ruler, tok2vec, merge_noun_chunks, merge_entities, merge_subtokens, token_splitter, doc_cleaner, parser, beam_parser, lemmatizer, trainable_lemmatizer, entity_linker, ner, beam_ner, entity_ruler, tagger, morphologizer, senter, sentencizer, textcat, spancat, future_entity_ruler, span_ruler, textcat_multilabel, en.lemmatizer
```

Thank you very much in advance for any help.

Additional info:
Python 3.7.4
spacy==3.3.1
spacy-legacy==3.0.9
spacy-loggers==1.0.3
en-core-web-sm @ https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.3.0/en_core_web_sm-3.3.0-py3-none-any.whl

**Comment by Lucaterre** [comment]
Hi @angelosalatino ,

I think it is an error in the documentation, have you tried with Python 3.8 or Python 3.9? Just to know, what version of spacyfishing are you using? (I don't see it in the requirements ("Additional info") that you provide me).

thank in advance

**Comment by angelosalatino** [comment]
Hi @Lucaterre,

I used python 3.8, and it is going.

Previously, when the error came up, I was using python 3.7. I was using pip install, and it wouldn't do anything. It returned me and error. I guess because it requires python 3.8.

In this case, I went to follow the developer version. But the following excerpt:
```
git clone https://github.com/Lucaterre/spacyfishing.git
virtualenv --python=/usr/bin/python3.8 venv
source venv/bin/activate
pip install -r requirements_dev.txt
```
was not enough.

It wouldn't have worked anyway, but I guess at this stage it needs:
```
cd spacyfishing
pip install ./
```

But I am not 100% sure, as then I went to 3.8 and it worked.

**Comment by Lucaterre** [comment]
Great if it work with python 3.8 :+1:

For the version problem, It seems the origin comes from a bad error in `setup.py`: the Python version required is set to `>= 3.8` and not `>= 3.7`, and that's why it causes an error with pip and version 3.7. I'll fix that and release soon.

With dev installation, module import is required in the script (you have an example with the `tests/` folder), which is not the case for normal installation.

Thank you for this feedback.

**Comment by Lucaterre** [comment]
The `0.1.7` release fixes the installation with Python 3.7 with pip, I close the topic (feel free to open it again, if a problem comes back).
